src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.cs

```diff
@@ -504,38 +505,45 @@ public static void ExtractToDirectory(String sourceArchiveFileName, String desti
                 // to be greater than the length of typical entry names from the file system, even
                 // on non-Windows platforms. The capacity will be increased, if needed.
                 const int DefaultCapacity = 260;
-                char[] entryNameBuffer = new char[DefaultCapacity];
+                char[] entryNameBuffer = ArrayPool<char>.Shared.Rent(DefaultCapacity);
```

Comment:
This is being passed by ref to EntryFromPath, which passes it by ref to EnsureCapacity, which may throw it away and allocate a different buffer.  Isn't that potentially an issue, that this buffer isn't being returned and that a different one may be?